webapp/src/ts/training-card.guard.provider.ts

```diff
@@ -0,0 +1,41 @@
+import { Injectable } from '@angular/core';
+import { CanDeactivate, RouterStateSnapshot, ActivatedRouteSnapshot } from '@angular/router';
+import { Store } from '@ngrx/store';
+import { Observable, map, take } from 'rxjs';
+
+import { GlobalActions } from '@mm-actions/global';
+import { Selectors } from '@mm-selectors/index';
+
+@Injectable({
+  providedIn: 'root'
+})
+export class TrainingCardDeactivationGuardProvider implements CanDeactivate<any> {
+  private globalActions;
+
+  constructor(private store: Store) {
+    this.globalActions = new GlobalActions(this.store);
+  }
+
+  canDeactivate(
```

Comment:
[The privacy policy route also needs it ](https://github.com/medic/cht-core/blob/master/webapp/src/ts/modules/privacy-policy/privacy-policy.routes.ts)